cmd/backup/script.go

```diff
@@ -107,112 +99,9 @@ func newScript() (*script, error) {
 		s.cli = cli
 	}
 
-	if s.c.AwsS3BucketName != "" {
```

Comment:
I would prefer if we could keep all logic that conditionally creates a storage client or not based on the configuration in `func newScript`.

There's a comment

```
// All reading from env vars or other configuration sources is expected to happen
// in this method.
```

and I think it still makes sense to do it like this.

Each block could then add the storage to the pool or not.

Reply:
Hmm I get your point. A storage backend should have the authority to decide itself whether it shall be initialized. If it cannot do that even though it is asked to, it can still fail. While that may create duplicate code (check if should init in script and then check presence of partly the same configs again), it makes more logical sense, I agree.